dataRetriever/resolvers/miniblockResolver.go

```diff
@@ -0,0 +1,202 @@
+		return nil, dataRetriever.ErrNilAntifloodHandler
+	}
+	if check.IfNil(arg.Throttler) {
+		return nil, dataRetriever.ErrNilThrottler
+	}
+	if check.IfNil(arg.DataPacker) {
+		return nil, dataRetriever.ErrNilDataPacker
+	}
+
+	mbResolver := &miniblockResolver{
+		TopicResolverSender: arg.SenderResolver,
+		miniBlockPool:       arg.MiniBlockPool,
+		miniBlockStorage:    arg.MiniBlockStorage,
+		dataPacker:          arg.DataPacker,
+		messageProcessor: messageProcessor{
+			marshalizer:      arg.Marshalizer,
+			antifloodHandler: arg.AntifloodHandler,
+			topic:            arg.SenderResolver.RequestTopic(),
+			throttler:        arg.Throttler,
+		},
+	}
+
+	return mbResolver, nil
+}
+
+// ProcessReceivedMessage will be the callback func from the p2p.Messenger and will be called each time a new message was received
+// (for the topic this validator was registered to, usually a request topic)
+func (mbRes *miniblockResolver) ProcessReceivedMessage(message p2p.MessageP2P, fromConnectedPeer p2p.PeerID) error {
+	err := mbRes.canProcessMessage(message, fromConnectedPeer)
+	if err != nil {
+		return err
+	}
+
+	mbRes.throttler.StartProcessing()
+	defer mbRes.throttler.EndProcessing()
+
+	rd, err := mbRes.parseReceivedMessage(message)
+	if err != nil {
+		return err
+	}
+
+	switch rd.Type {
+	case dataRetriever.HashType:
+		var buff []byte
+		buff, err = mbRes.resolveMbRequestByHash(rd.Value)
+		if err != nil {
+			return err
+		}
+		return mbRes.Send(buff, message.Peer())
+	case dataRetriever.HashArrayType:
+		return mbRes.resolveMbRequestByHashArray(rd.Value, message.Peer())
+	default:
+		return dataRetriever.ErrRequestTypeNotImplemented
+	}
+}
+
+func (mbRes *miniblockResolver) resolveMbRequestByHash(hash []byte) ([]byte, error) {
+	mb, err := mbRes.fetchMbAsByteSlice(hash)
+	if err != nil {
+		return nil, err
+	}
+
+	buffToSend, err := mbRes.marshalizer.Marshal(&batch.Batch{Data: [][]byte{mb}})
+	if err != nil {
+		return nil, err
+	}
+
+	return buffToSend, nil
+}
+
+func (mbRes *miniblockResolver) fetchMbAsByteSlice(hash []byte) ([]byte, error) {
+	value, ok := mbRes.miniBlockPool.Peek(hash)
+	if ok {
+		txBuff, err := mbRes.marshalizer.Marshal(value)
+		if err != nil {
+			return nil, err
+		}
+		return txBuff, nil
+	}
+
+	return mbRes.miniBlockStorage.SearchFirst(hash)
+}
+
+func (mbRes *miniblockResolver) resolveMbRequestByHashArray(mbBuff []byte, pid p2p.PeerID) error {
+	b := batch.Batch{}
+	err := mbRes.marshalizer.Unmarshal(&b, mbBuff)
+	if err != nil {
+		return err
+	}
+	hashes := b.Data
+
+	mbsBuffSlice := make([][]byte, 0, len(hashes))
+	for _, hash := range hashes {
+		mb, errFetch := mbRes.fetchMbAsByteSlice(hash)
+		if errFetch != nil {
+			err = errFetch
+			log.Trace("fetchTxAsByteSlice missing",
+				"error", err.Error(),
+				"hash", hash)
+			continue
+		}
+		mbsBuffSlice = append(mbsBuffSlice, mb)
+	}
+
+	buffsToSend, errPack := mbRes.dataPacker.PackDataInChunks(mbsBuffSlice, maxBuffToSendBulkTransactions)
```

Comment:
Why use the constant set for transactions ? Maybe rename it so it is more general

Reply:
good point! added a new const for this

Reply:
I think we will leave it for now to 256k, will change these if we found some better values